src/cats/util.cljc

```diff
@@ -36,3 +42,14 @@
              IPrintWithWriter
              (-pr-writer [mv writer _]
                (-write writer (cats.protocols/-repr mv))))))
+
+(defmacro mapl [f coll]
```

Comment:
It's similarly not clear to me why `mapl` is defined as a macro. This should live in `builtin.cljc` too.

Maybe I'm missing something, but why not something like this instead?

```clj
(defn- mapl [f lst] (reverse (reduce #(cons (f %2) %1) () coll)))
```

N.B. There's no need to quote the empty list.

Pedantic: The possessive `'s` is misused here and in `flattenl-1`.